### PR TITLE
WIP Customer birthday

### PR DESCRIPTION
**Pull request by gillesbourgeat**
Add customer birthday in : 
- model, insert and update
- loop customer
- form register
- form customer update account
- form admin customer edit
- export customer

Configuration variables : 
- customer_birthday_enable for active birthday
- customer_birthday_required for force to the birthday

The birthday format depends on the country date format

Cheers,

### CODE REVIEW

File: core/lib/Thelia/Form/CustomerUpdateForm.php [hunk 1]
```diff
@@ -202,4 +220,24 @@ public function verifyEmailField($value, ExecutionContextInterface $context)
             $context->addViolation(Translator::getInstance()->trans("email confirmation is not the same as email field"));
         }
     }
+
+    /**
+     * @return string
+     */
+    private function getFormatBirthday()
+    {
+        if ($this->getRequest()->getSession() != null) {
```
Comment (lovenunu) [inline core/lib/Thelia/Form/CustomerUpdateForm.php]: Whoooo, seems like a copy/paste ;)
Why didn't you do a service that handle this formatter ?

File: setup/insert.sql [hunk 2]
```diff
@@ -155,6 +158,8 @@ INSERT INTO `config_i18n` (`id`, `locale`, `title`, `description`, `chapo`, `pos
 (36, 'fr_FR', 'Durée de vie du cookie de la session dans le navigateur du client, en secondes', NULL, NULL, NULL),
 (37, 'fr_FR', 'Permettre aux utilisateurs de changer leur email. 1 pour oui, 0 pour non', NULL, NULL, NULL),
 (38, 'fr_FR', 'Demander aux clients de confirmer leur email. 1 pour oui, 0 pour non', NULL, NULL, NULL)
```
Comment (lunika) [inline setup/insert.sql]: comma is missing at the end of this line.

File: core/lib/Thelia/Controller/Admin/CustomerController.php [hunk 3]
```diff
@@ -18,6 +18,7 @@
 use Thelia\Core\Security\Resource\AdminResources;
 use Thelia\Exception\CustomerException;
 use Thelia\Form\Definition\AdminForm;
+use Thelia\Model\Customer;
```
Comment (lunika) [inline core/lib/Thelia/Controller/Admin/CustomerController.php]: This class seems never used.

**Comment by lunika** [comment]
Complete the customer test suite adding test test when birthday date is enabled.

Thanks.

**Comment by gillesbourgeat** [comment]
:two_hearts: Rebase the infinity :two_hearts:

**Pull request closed without merging**